### PR TITLE
[redux-form] Export BaseFieldProps

### PR DESCRIPTION
**Pull request by doomsower**
This PR just exports BaseFieldProps to allow this style of using custom fields:

```jsx
<FieldCustomComp name="field_4_comp" foo="bar" />
```

instead (in addition) of this:
```jsx
<FieldCustom name="field4" component={ MyField } foo="bar" />
```
I prefer this style because it is more concise, and if you declare `FieldCustom` you are unlikely to use it with component other than `MyField` anyway, so why not to take one more step and declare `FieldCustomComp`.
 

Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by doomsower** [comment]
@carsonf @aikoven @LKay @bancek @alsiola
For some reason dt-bot ignored this PR
Do you approve it?

**Review by tehbi4 — APPROVED** [review]
LGTM!

**Comment by RyanCavanaugh** [comment]
Approved by third party. PR appears ready to merge pending review by a maintainer.

**Pull request merged**
